signs-work-order-map/components/Map.tsx

```diff
@@ -159,6 +202,45 @@ export default function Map({ signs, messageType, editLocation }: MapProps) {
         <SignPopup popupInfo={popupInfo} setPopupInfo={setPopupInfo} />
       )}
 
+      {/* Loading indicator for AGOL data */}
+      {agolLoading && (
+        <div
+          style={{
+            position: "absolute",
+            top: "10px",
+            right: "10px",
+            background: "rgba(255, 255, 255, 0.9)",
+            padding: "8px 12px",
+            borderRadius: "4px",
+            fontSize: "14px",
+            boxShadow: "0 2px 4px rgba(0,0,0,0.1)",
+            zIndex: 1000,
+          }}
+        >
+          Loading sign assets...
+        </div>
```

Comment:
I had to throttle down my network to get to actually see this, but I'm glad I did -- it looks good!